### DISCUSSION TITLE
inittree does not invoke init()

### DISCUSSION THREAD

**Discussion by ericxin1982** [alpinejs/alpine#4163] [1. Help]
![image](https://github.com/alpinejs/alpine/assets/9047436/3feb5385-0e62-495e-98de-d2ac09f08dd5)

![image](https://github.com/alpinejs/alpine/assets/9047436/43ba7644-5c86-4ffd-bb0f-a82dd0ac9992)

I am doing custom element programming, I try to make each code separately as much as possible, I learned this code writing from google, it works but only init() function inside object it does not invoke, anyone know why?

Thanks

Eric Xin

**Comment by SimoTod** [comment]
Init being autocalled is a feature of the x-data directive. You are using a low level api, addScopeToNode, which is just attaching some data to an element. Basically you are trying to rewrite part of the Alpine core.

**Comment by SimoTod** [comment]
The code is open source, if you try to do something using low level APIs, you have to look at the existing code and understand how it works first.

**Reply by ericxin1982** [reply]
I can guess the function name and its usage, I just try to assembly the code tidy and easily reuse, I have no plan to rewrite the alpine core, only this writing have shadow dom available, I just wondering why this way init() does not work, and how to make it work like when it is inside x-data.

**Reply by SimoTod** [reply]
Even if it's not your plan, it's what you are doing :D
Your implementation will not support magics either, x-data does other stuff as well, see https://github.com/alpinejs/alpine/blob/main/packages/alpinejs/src/directives/x-data.js#L13

My advice is to use the documentation, if it's not documented, read the code.
Otherwise you'll be guessing wrong, have bugs and then end up asking questions but, because you are not using the framework in a canonical way, not many people will be able to help/answer.

**Reply by ericxin1982** [reply]
https://github.com/alpinejs/alpine/issues/172
![image](https://github.com/alpinejs/alpine/assets/9047436/4bb2edae-9ac7-4b89-8bfb-636d671fa535)
I think I just continue this attempt, and shadow dom is running good without invoke init, I am just curious about how to solve.

Thanks
Eric Xin

**Comment by ericxin1982** [comment]
I think I have bypassed this concern. Thanks all for your help!
